src/app/files/fileManager.js

```diff
@@ -136,30 +137,57 @@ class FileManager extends FileSystemApi {
   async setFile (path, content) {
     if (this.currentRequest) {
       let reject = false
-      let savedAsAnotherFile = false
+      let saveAsCopy = false
       let actions = (toaster) => {
-        return yo`<div class="container ml-1">
-        <button class="btn btn-primary btn-sm m-1" onclick=${(e) => { reject = true; e.target.innerHTML = 'Canceled'; toaster.hide() }}>Cancel</button>
-        <button class="btn btn-primary btn-sm m-1" onclick=${(e) => {
-          if (savedAsAnotherFile) return
-          savedAsAnotherFile = true
-          const newPath = path + '.' + this.currentRequest.from
-          this._setFileInternal(newPath, content)
-          this.switchFile(newPath)
-          e.target.innerHTML = 'Saved'
-          toaster.hide()
-        }}>Save As Copy</button>
-        </div>`
+        return yo`
+          <div class="container ml-1">
+            <button class="btn btn-primary btn-sm m-1" onclick=${(e) => {
```

Comment:
Create a closure (local function) instead of writing the function inside the `onclick` tag